src/content/contribute/writing-a-plugin.md

```diff
@@ -3,125 +3,141 @@ title: Writing a Plugin
 sort: 4
 contributors:
   - tbroadley
+  - iamakulov
 ---
 
 Plugins expose the full potential of the webpack engine to third-party developers. Using staged build callbacks, developers can introduce their own behaviors into the webpack build process. Building plugins is a bit more advanced than building loaders, because you'll need to understand some of the webpack low-level internals to hook into them. Be prepared to read some source code!
 
 ## Creating a Plugin
 
-A plugin for `webpack` consists of
+A plugin for `webpack` consists of a named JavaScript class that:
 
-- A named JavaScript function.
-- Defines `apply` method in its prototype.
+- Defines the `apply` method.
 - Specifies an [event hook](/api/compiler-hooks/) on which to bind itself.
 - Manipulates webpack internal instance specific data.
-- Invokes webpack provided callback after functionality is complete.
 
 ```javascript
-// A named JavaScript function.
-function MyExampleWebpackPlugin() {
-
-};
-
-// Defines `apply` method in it's prototype.
-MyExampleWebpackPlugin.prototype.apply = function(compiler) {
-  // Specifies webpack's event hook to attach itself.
-  compiler.plugin('webpacksEventHook', function(compilation /* Manipulates webpack internal instance specific data. */, callback) {
-    console.log("This is an example plugin!!!");
-
-    // Invokes webpack provided callback after functionality is complete.
-    callback();
-  });
-};
+// A named JavaScript class...
+class MyExampleWebpackPlugin {
+  // ...that defines `apply` method in its prototype...
```

Comment:
Can we change some of this comments? I suggest removing the first one since its redundant. Be more imperative, like _Define apply method in the class._ (without trailing periods and more connected with the requirements)